### PR TITLE
Migrate to ANTLR 4.6.4

### LINKED ISSUE #2498: Upgrade to Antlr 4.6
We're currently running off Antlr 4.3; Antlr 4.6 has important performance improvements that RD will massively benefit from, but upgrading isn't going to work if we merely swap the library - we need to rebuild *every single parser* with 4.6.

- Rubberduck.Parsing.Grammar has **VBALexer.g4** and **VBAParser.g4**.
- Rubberduck.Parsing.Preprocessing has **VBAConditionalCompilationParser.g4**, **VBADate.g4**, and **VBALike.g4**.

When 4.6 was released I rebuilt VBALexer and VBAParser, but forgot about the Preprocessing grammars; that resulted in several hundred broken tests.

### PR DESCRIPTION
**Pull request by Vogel612**
Delete all the manually generated codefiles and add build-actions for all *.g4 files that can update all our stuff.
Furthremore VBALexer.g4 and VBAParser.g4 have been reencoded as UTF-8 files.

The Lexer rules `B_CHAR` and `BF` have been deprecated in favor of Semantic Predicates.

Fixes #2498

### CODE REVIEW

**Comment by Vogel612** [comment]
currently somewhat blocked by ANTLR being weird. more information in https://stackoverflow.com/questions/48372468/parsing-letter-ranges-with-antlr

**Comment by Vogel612** [comment]
TBD: Rewrite the heck out of B_CHAR, BF and everything that uses `RESTRICTED_LETTER`.

All these lexer rules need to die and be replaced by predicates that match over an `unrestrictedIdentifier`

File: RetailCoder.VBE/Refactorings/ReorderParameters/ReorderParametersRefactoring.cs [hunk 1]
```diff
@@ -151,7 +151,7 @@ private void RewriteCall(VBAParser.ArgumentListContext argList, ICodeModule modu
             var args = argList.argument().Select((s, i) => new { Index = i, Text = s.GetText() }).ToList();
             for (var i = 0; i < _model.Parameters.Count; i++)
             {
-                if (argList.argument().Count <= i)
+                if (argList.argument().Count() <= i)
```
Comment (retailcoder) [inline RetailCoder.VBE/Refactorings/ReorderParameters/ReorderParametersRefactoring.cs]: If that's an array, better use its `Length` property. If it's not an array, ignore this comment :)

File: Rubberduck.Parsing/Preprocessing/DateLiteralExpression.cs [hunk 2]
```diff
@@ -71,7 +71,7 @@ public override IValue Evaluate()
                 var L = dateValue.dateValuePart()[0];
                 var M = dateValue.dateValuePart()[1];
                 VBADateParser.DateValuePartContext R = null;
-                if (dateValue.dateValuePart().Count == 3)
+                if (dateValue.dateValuePart().Length == 3)
```
Comment (retailcoder) [inline Rubberduck.Parsing/Preprocessing/DateLiteralExpression.cs]: Seems they *are* arrays after all

File: Rubberduck.Parsing/TokenExtensions.cs [hunk 3]
```diff
@@ -15,7 +15,7 @@ public static class TokenExtensions
         /// <returns>Zero-based column position</returns>
         public static int EndColumn(this IToken token)
         {
-            if (token.Text == Environment.NewLine)
+            if (token.Text == Environment.NewLine || token.Type == TokenConstants.Eof)
```
Comment (retailcoder) [inline Rubberduck.Parsing/TokenExtensions.cs]: :+1:

**Pull request merged**